packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -282,60 +311,122 @@ class _ZoomEnterTransition extends StatelessWidget {
+        ? _scaleDownTransition
+        : _scaleUpTransition
+      ).animate(widget.animation);
+
+    widget.animation.addListener(onAnimationValueChange);
+    widget.animation.addStatusListener(onAnimationStatusChange);
+  }
+
   @override
-  Widget build(BuildContext context) {
-    double opacity = 0;
-    // The transition's scrim opacity only increases on the forward transition.
-    // In the reverse transition, the opacity should always be 0.0.
-    //
-    // Therefore, we need to only apply the scrim opacity animation when
-    // the transition is running forwards.
-    //
-    // The reason that we check that the animation's status is not `completed`
-    // instead of checking that it is `forward` is that this allows
-    // the interrupted reversal of the forward transition to smoothly fade
-    // the scrim away. This prevents a disjointed removal of the scrim.
-    if (!reverse && animation.status != AnimationStatus.completed) {
-      opacity = _scrimOpacityTween.evaluate(animation)!;
-    }
+  void initState() {
+    _updateAnimations();
+    delegate = _ZoomEnterTransitionDelegate(
+      reverse: widget.reverse,
+      fade: fadeTransition,
+      scale: scaleTransition,
+      animation: widget.animation,
+    );
+    super.initState();
+  }
 
-    final Animation<double> fadeTransition = reverse
-      ? kAlwaysCompleteAnimation
-      : _fadeInTransition.animate(animation);
+  @override
+  void didUpdateWidget(covariant _ZoomEnterTransition oldWidget) {
+    if (oldWidget.reverse != widget.reverse || oldWidget.animation != widget.animation) {
+      oldWidget.animation.removeListener(onAnimationValueChange);
+      oldWidget.animation.removeStatusListener(onAnimationStatusChange);
+      _updateAnimations();
+      delegate.dispose();
+      delegate = _ZoomEnterTransitionDelegate(
+        reverse: widget.reverse,
+        fade: fadeTransition,
+        scale: scaleTransition,
+        animation: widget.animation,
+      );
+    }
+    super.didUpdateWidget(oldWidget);
+  }
 
-    final Animation<double> scaleTransition = (reverse
-      ? _scaleDownTransition
-      : _scaleUpTransition
-    ).animate(animation);
+  @override
+  void dispose() {
+    widget.animation.removeListener(onAnimationValueChange);
+    widget.animation.removeStatusListener(onAnimationStatusChange);
+    delegate.dispose();
+    super.dispose();
+  }
 
+  @override
+  Widget build(BuildContext context) {
+    if (allowRasterization) {
+      return RasterWidget(
+        delegate: delegate,
+        controller: controller,
+        fallback: delegate,
+        child: widget.child,
+      );
+    }
     return AnimatedBuilder(
-      animation: animation,
+      animation: widget.animation,
       builder: (BuildContext context, Widget? child) {
+        double scrimOpacity = 0;
+        // The transition's scrim opacity only increases on the forward transition.
+        // In the reverse transition, the opacity should always be 0.0.
```

Comment:
Can you assert here that the 0.0 is true?

Reply:
Done